WordPress/Classes/ViewRelated/EEUUSCompliance/CompliancePopoverViewModel.swift

```diff
@@ -5,19 +5,30 @@ import WordPressUI
 final class CompliancePopoverViewModel: ObservableObject {
     @Published
     var isAnalyticsEnabled: Bool = !WPAppAnalytics.userHasOptedOut()
+
+    // MARK: - Dependencies
+
+    let analyticsTracker: PrivacySettingsAnalyticsTracking
```

Comment:
[Suggestion] I think making this private and exposing a function that handles tracking the display of the popover is better for encapsulation. The function could be named `didDisplayPopover`.

Reply:
This will also make writing tests easier since we can directly test the view model, and there won't be a clear need for `CompliancePopoverViewControllerTests`. What do you think?

Reply:
Agreed, I wasn't a fan of exposing the `analyticsTracker` property. Thanks for the suggestion.

Reply:
> This will also make writing tests easier since we can directly test the view model, and there won't be a clear need for CompliancePopoverViewControllerTests. What do you think?

@salimbraksa Any thoughts on this?

Reply:
>Imagine there is a change that accidentally removes the viewModel call from the viewController.viewDidLoad, in this case, the viewModel tests will pass, even though the feature is not working.
...
> I'm not saying we shouldn't test the viewModel but even if we do, we should at least write a unit test that checks whether the viewController calls the viewModel.didDisplayPopover method.

You pose a valid point. However, since `CompliancePopoverViewControllerTests` currently only tests one tracking flow, I believe adding tests to `CompliancePopoverViewModelTests` is important. We track three different events, and we can easily test those flows through the presenter.

Reply:
Resolved in 2657b503b7a905216de3339c381288fafd69f5b8